[ADD] Studio: New webhooks doc (#12255)
odoo/documentation: content/applications/studio/automated_actions/webhooks.rst

to the webhook's URL.

.. danger::
   The webhook's URL is **confidential** and should be treated with care. Sharing it online or
   without caution can provide unintended access to the Odoo database. Click :guilabel:`Rotate
   Secret` to change the URL if needed.

.. _studio/automations/webhook_action:

Set the webhook's action
------------------------

Next, set the webhook's actions. This is what happens when the webhook's URL is called. Ensure the
correct Odoo fields are targeted, whether updating or computing, and that the action type is
configured correctly. To check that the webhook was properly set up, make sure to :ref:`test it
<studio/automations/test_webhook>`.

To set a webhook's action while configuring a webhook, click :guilabel:`Add an action` under the
:guilabel:`Actions To Do` tab, and then click the action's :guilabel:`Type` (what it does when it
receives a payload), select what :guilabel:`Allowed Groups` of users can send data to this webhook,
and then set the :guilabel:`Action Details`. To see an example of how a webhook action could be set
up, :ref:`review these example use cases <studio/automations/webhook_examples>`.

> .. important::
>    Although the :guilabel:`Model` is set in Odoo, its payload format differs and must be included in
>    the payload. Click the model's internal link to find the correct format in the Model field. For
>    example, a sales order webhook uses the `Sales Order` model but has a `sale.order` payload
>    format.
> 
>    .. image:: webhooks/webhook-sales-order-model.png
>       :alt: The sales order model for a webhook's payload.

Reviewer: I'm a bit confused here. When I worked through the test use cases, I didn't see anywhere I had to add the technical name of the model. 

What do we expect the user to do with the technical name? (Also, the user needs to turn on developer mode to be able to click that 'Internal link' so I would update that ... since I think the rest should be updated to explain how to set this up not using developer mode.)

```suggestion
.. important::
   Although the :guilabel:`Model` is set in Odoo, it is the model's technical name that must be included in the payload. Click the model's internal link to find this name in the Model field. For
   example, a sales order webhook uses the *Sales Order* model but the technical name `sale.order` is used in the payload.
```

<studio/automations/webhook_trigger>` so the :guilabel:`Model` is set to `Sales Order`. Also, set
the :guilabel:`Target Record` to `model.env[payload.get('model')].browse(int(payload.get('id')))`.
This is broken down below.

- **model**: What gets updated in Odoo (in this case, a sales order). This matches the
  :guilabel:`Model` set earlier.
- **env**: Where the action takes place. In this case, it is Odoo.
- **payload**: This is what is sent to the webhook's URL. It contains the information that updates
  the sales order.
- **get('model')**: Tells the webhook what database record to look at. In this case, the webhook
  retrieves (`get`) the data tied to a specific record `model`. In this example, this is the sales
  orders.
- **browse**: This tells the webhook to look in the `model` (the sales orders) set by the payload
  for what to update.
- **int**: This turns the target into an `integer` (a whole number). This is important in case some
  words (a `string`) or a decimal number is included in the payload's target record.
- **get('id')**: Identifies the sales order number that is getting updated in Odoo.

Set the webhook's action
~~~~~~~~~~~~~~~~~~~~~~~~

> After setting the trigger, set the webhook's action by clicking :guilabel:`Add an action`. For the
> :guilabel:`Type`, click :guilabel:`Update Record`. And then, set the :guilabel:`Action Details` to
> `Update Customer to USD`. Finally, click :guilabel:`Save & Close`.

Reviewer: I'm not 100 sure what the best approach is for formatting what the user is setting here in the Actions To Do. Right now though, it looks like it's one single 'value' that needs to be entered, which is not the case. See my suggestion below, which is similar to how Audrey did it on the `[main Automated actions page](https://www.odoo.com/documentation/18.0/applications/studio/automated_actions.html#update-record)`. Note that 'Action details' is only visible in developer mode, so edit accordingly.

```suggestion
After setting the trigger, set the webhook's action by clicking :guilabel:`Add an action`. For the
:guilabel:`Type`, click :guilabel:`Update Record`. Then, select `Update`, choose the field `Currency` then select `USD` to have the currency field updated to USD. Finally, click :guilabel:`Save & Close`.
```

Webhook setup summary
~~~~~~~~~~~~~~~~~~~~~

To summarize what is set up, the webhook targets sales orders, identified by their sales order
number, and updates their currency to `USD` when an API call is sent to the webhook's URL that
includes that sales order number (which is identified by the payload's `id` record).

Test the webhook
~~~~~~~~~~~~~~~~

Test the webhook's setup to make sure everything is correct. This process uses a tool called
`Postman <https://www.postman.com/>`_ to send the simulated trigger.

This section walks through the steps to test this webhook in Postman, but does not offer help if
there's an issue within that tool. To get specific help with Postman, contact their support team.

> Once Postman is open, create a new :guilabel:`HTTP` request, and set its method to :guilabel:`POST`.

Reviewer: I think a numbered list would work well here for lines 202 to 218.

Next, copy the webhook's URL that is being tested and paste it into the URL field in Postman. After
that, click the :guilabel:`Body` tab and select the :guilabel:`raw` option. Set the file type to
:guilabel:`JSON`, then copy this code and paste it into the file.

.. code-block:: json

   {
       "model": "sale.order",
       "id": "SALES ORDER NUMBER"
   }

From here, pick a sales order to test the sales order on. If it is not possible to test in a live
Odoo database, consider creating a demo database with a sample sales order and the webhook that was
configured. Replace `SALES ORDER NUMBER` with the sales order's number without the `S` or any zeros
before the number. For example, a sales order with the number `S00007` should be entered as `7` when
it is entered into Postman. Finally, click :guilabel:`Send` in Postman.

> If a message saying `200 OK` or `status: ok` gets returned, then the webhook is functioning properly

Reviewer: I would look for a way to avoid repeating the info from lines 220 to 227? This is essentially repeated three times on the page.

(Similar comments for the second example use case.)
